src/graphics/device.js

```diff
@@ -1511,6 +1511,43 @@ Object.assign(pc, function () {
             }
         },
 
+        // In the case where a texture has more than 1 level of mip data specified, but doesn't have
+        // the full mip chain, we generate the missing levels here.
+        // This is to overcome an issue where iphone xr and xs ignores further updates to the mip data
+        // after invoking gl.generateMipmap on the texture (which was the previous method of ensuring
+        // the texture's full mip chain was complete).
+        // NOTE: this function copies texture data from the previous mip level instead of downsampling it.
```

Comment:
Since the missing mips shouldn't be used in the IBL code, what's the difference between copying texture data and just providing an array of black pixels? And, for that matter, what's the difference between copying texture data and supplying beautifully downsampled mip images.

Reply:
This code applies to all textures, not just IBL cubemaps. So it's possible there exist non IBL assets that could be impacted. Also, I believe the lower mips may be accessed by the card when performing the trilinear blend (though this may be small enough to completely ignore).

I could go ahead and blank out the mipmap data, but this would actually be an extra step in the process and take longer than the above. (Unless there's a better way to create the typed array than using subarray()).

Reply:
OK cool. But how about "what's the difference between copying texture data and supplying beautifully downsampled mip images."?

Reply:
I mean, we _could_ nicely downsample.

Reply:
For IBL, here is the difference between copying random pixels and blackening the last 2 levels of mipmap:
<img width="250" alt="mips" src="https://user-images.githubusercontent.com/11276292/58644559-6a18f980-82f9-11e9-9ab1-88d2a0fd678c.png">

Reply:
We could nicely downsample, but it wouldn't be straightforward to support all the various possible pixel formats. I actually implemented downsample for rgba8... perhaps we could actually use that in the (presumably common) case of rgba8 and fallback to pixel copy for the other formats?